internal/antewrapper/tracing_meter.go

```diff
@@ -0,0 +1,83 @@
+package antewrapper
+
+import (
+	"fmt"
+
+	"github.com/armon/go-metrics"
+	"github.com/tendermint/tendermint/libs/log"
+
+	sdkgas "github.com/cosmos/cosmos-sdk/store/types"
+	"github.com/cosmos/cosmos-sdk/telemetry"
+)
+
+type tracingGasMeter struct {
+	// a context logger reference for info/debug output
+	log log.Logger
+	// the gas meter being wrapped
+	base sdkgas.GasMeter
+	// tracks amount used per purpose
+	used map[string]uint64
+	// tracks number of usages per purpose
+	calls map[string]uint64
+}
+
+// NewTracingMeterWrapper returns a reference to a new tracing gas meter that will track calls to the base gas meter
+func NewTracingMeterWrapper(logger log.Logger, baseMeter sdkgas.GasMeter) sdkgas.GasMeter {
+	return &tracingGasMeter{
+		log:   logger,
+		base:  baseMeter,
+		used:  make(map[string]uint64),
+		calls: make(map[string]uint64),
+	}
+}
+
+// GasConsumed reports the amount of gas consumed at Log.Info level
```

Comment:
Do we want an interface check here?
```suggestion
var _  sdkgas.GasMeter = &tracingGasMeter{}

// GasConsumed reports the amount of gas consumed at Log.Info level
```

Reply:
Wouldn’t hurt